lang/cs/Org.Apache.REEF.Network/Group/Driver/Impl/CommunicationGroupDriver.cs

```diff
@@ -133,17 +134,30 @@ public class CommunicationGroupDriver : ICommunicationGroupDriver
         /// <param name="masterTaskId">The master task id in broadcast operator</param>
         /// <param name="codecType">The Codec used for serialization</param>
         /// <param name="topologyType">The topology type for the operator</param>
+        /// <param name="pipelineDataConverter">The class type used to convert data back and forth to pipelined one</param>
         /// <returns>The same CommunicationGroupDriver with the added Broadcast operator info</returns>
-        public ICommunicationGroupDriver AddBroadcast<T>(string operatorName, string masterTaskId, ICodec<T> codecType, TopologyTypes topologyType = TopologyTypes.Flat)
+        public ICommunicationGroupDriver AddBroadcast<T>(string operatorName, string masterTaskId, ICodec<T> codecType, TopologyTypes topologyType = TopologyTypes.Flat, IPipelineDataConverter<T> pipelineDataConverter = null)
         {
             if (_finalized)
             {
                 throw new IllegalStateException("Can't add operators once the spec has been built.");
             }
 
-            var spec = new BroadcastOperatorSpec<T>(
-                masterTaskId,
-                codecType);
+            BroadcastOperatorSpec<T> spec;
+
+            if (pipelineDataConverter == null)
```

Comment:
You don't need if else here, always call BroadcastOperatorSpec constructor with pipelineDataConverter, if pipelineDataConverter is null, it will be set as null in BroadcastOperatorSpec constructor anyway.